apiserver/facades/client/application/application.go

```diff
@@ -2236,9 +2239,46 @@ func (api *APIBase) DeployFromRepository(ctx context.Context, args params.Deploy
 	}, nil
 }
 
-func (api *APIBase) getOneApplicationStorage(entity params.Entity) (map[string]params.StorageDirectives, error) {
-	// TODO(storage): implement and add test.
-	return nil, errors.NotImplementedf("GetApplicationStorage")
+func (api *APIBase) getOneApplicationStorage(ctx context.Context, entity params.Entity) (map[string]params.StorageDirectives, error) {
+	appTag, err := names.ParseTag(entity.Tag)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	appUUID, err := api.applicationService.GetApplicationUUIDByName(ctx, appTag.Id())
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	storageDirectives, err := api.applicationService.GetApplicationStorageDirectives(ctx, appUUID)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	sc := make(map[string]params.StorageDirectives)
```

Comment:
Pre-allocate this. We know how big it has to be. `make(map[string]params.StorageDirectives, len(storageDirectives`